core/dbt/dataclass_schema.py

```diff
@@ -1,97 +1,133 @@
+    }
+    serialize_by_alias = True
+
+
+# This class pulls in DataClassDictMixin from Mashumaro. The 'to_dict'
+# and 'from_dict' methods come from Mashumaro.
+class dbtClassMixin(DataClassDictMixin):
     """The Mixin adds methods to generate a JSON schema and
     convert to and from JSON encodable dicts with validation
     against the schema
     """
 
-    class Config(MashBaseConfig):
-        code_generation_options = [
-            TO_DICT_ADD_OMIT_NONE_FLAG,
-        ]
-        serialization_strategy = {
-            datetime: DateTimeSerialization(),
-        }
+    _mapped_fields: ClassVar[Optional[Dict[Any, List[Tuple[Field, str]]]]] = None
+
+    # Config class used by Mashumaro
+    class Config(dbtMashConfig):
+        pass
 
-    _hyphenated: ClassVar[bool] = False
     ADDITIONAL_PROPERTIES: ClassVar[bool] = False
 
-    # This is called by the mashumaro to_dict in order to handle
-    # nested classes.
-    # Munges the dict that's returned.
-    def __post_serialize__(self, dct):
-        if self._hyphenated:
-            new_dict = {}
-            for key in dct:
-                if "_" in key:
-                    new_key = key.replace("_", "-")
-                    new_dict[new_key] = dct[key]
-                else:
-                    new_dict[key] = dct[key]
-            dct = new_dict
-
-        return dct
-
-    # This is called by the mashumaro _from_dict method, before
-    # performing the conversion to a dict
+    # This is called by the mashumaro from_dict in order to handle
+    # nested classes. We no longer do any munging here, but leaving here
+    # so that subclasses can leave super() in place for possible future needs.
     @classmethod
     def __pre_deserialize__(cls, data):
-        # `data` might not be a dict, e.g. for `query_comment`, which accepts
-        # a dict or a string; only snake-case for dict values.
-        if cls._hyphenated and isinstance(data, dict):
-            new_dict = {}
-            for key in data:
-                if "-" in key:
-                    new_key = key.replace("-", "_")
-                    new_dict[new_key] = data[key]
-                else:
-                    new_dict[key] = data[key]
-            data = new_dict
         return data
 
-    # This is used in the hologram._encode_field method, which calls
-    # a 'to_dict' method which does not have the same parameters in
-    # hologram and in mashumaro.
-    def _local_to_dict(self, **kwargs):
-        args = {}
-        if "omit_none" in kwargs:
-            args["omit_none"] = kwargs["omit_none"]
-        return self.to_dict(**args)
+    # This is called by the mashumaro to_dict in order to handle
+    # nested classes. We no longer do any munging here, but leaving here
+    # so that subclasses can leave super() in place for possible future needs.
+    def __post_serialize__(self, data):
+        return data
+
+    @classmethod
+    @functools.lru_cache
+    def json_schema(cls):
```

Comment:
Are the remaining methods here (lines 78-130) for backwards compatibility due to removing hologram classes and related methods? For example, are we keeping `clean_dictionary_keys_for_validate` because there's a concern that folks might have defined that downstream?

Reply:
Removed "clean_dictionary_keys_for_validate". I think it was an abandoned approach to some issue I've forgotten. The other methods are necessary and used in multiple places for validation, mostly of yaml dictionaries.